renderdoc/driver/metal/metal_stringise.cpp

```diff
@@ -0,0 +1,471 @@
+}
+
+template <>
+rdcstr DoStringise(const MTL::TessellationFactorFormat &el)
+{
+  BEGIN_ENUM_STRINGISE(MTL::TessellationFactorFormat)
+  {
+    MTL_STRINGISE_ENUM(TessellationFactorFormatHalf);
+  }
+  END_ENUM_STRINGISE()
+}
+
+template <>
+rdcstr DoStringise(const MTL::TessellationControlPointIndexType &el)
+{
+  BEGIN_ENUM_STRINGISE(MTL::TessellationControlPointIndexType)
+  {
+    MTL_STRINGISE_ENUM(TessellationControlPointIndexTypeNone);
+    MTL_STRINGISE_ENUM(TessellationControlPointIndexTypeUInt16);
+    MTL_STRINGISE_ENUM(TessellationControlPointIndexTypeUInt32);
+  }
+  END_ENUM_STRINGISE()
+}
+
+template <>
+rdcstr DoStringise(const MTL::TessellationFactorStepFunction &el)
+{
+  BEGIN_ENUM_STRINGISE(MTL::TessellationFactorStepFunction)
+  {
+    MTL_STRINGISE_ENUM(TessellationFactorStepFunctionConstant);
+    MTL_STRINGISE_ENUM(TessellationFactorStepFunctionPerPatch);
+    MTL_STRINGISE_ENUM(TessellationFactorStepFunctionPerInstance);
+    MTL_STRINGISE_ENUM(TessellationFactorStepFunctionPerPatchAndPerInstance);
+  }
+  END_ENUM_STRINGISE()
+}
+
+template <>
+rdcstr DoStringise(const MTL::TessellationPartitionMode &el)
+{
+  BEGIN_ENUM_STRINGISE(MTL::TessellationPartitionMode)
+  {
+    MTL_STRINGISE_ENUM(TessellationPartitionModePow2);
+    MTL_STRINGISE_ENUM(TessellationPartitionModeInteger);
+    MTL_STRINGISE_ENUM(TessellationPartitionModeFractionalOdd);
+    MTL_STRINGISE_ENUM(TessellationPartitionModeFractionalEven);
+  }
+  END_ENUM_STRINGISE()
+}
+
+template <>
+rdcstr DoStringise(const MTL::CPUCacheMode &el)
+{
+  BEGIN_ENUM_STRINGISE(MTL::CPUCacheMode)
+  {
+    MTL_STRINGISE_ENUM(CPUCacheModeDefaultCache);
+    MTL_STRINGISE_ENUM(CPUCacheModeWriteCombined);
+  }
+  END_ENUM_STRINGISE()
+}
+
+template <>
+rdcstr DoStringise(const MTL::StorageMode &el)
+{
+  BEGIN_ENUM_STRINGISE(MTL::StorageMode)
+  {
+    MTL_STRINGISE_ENUM(StorageModeShared);
+    MTL_STRINGISE_ENUM(StorageModeManaged);
+    MTL_STRINGISE_ENUM(StorageModePrivate);
+    MTL_STRINGISE_ENUM(StorageModeMemoryless);
+  }
+  END_ENUM_STRINGISE()
+}
+
+template <>
+rdcstr DoStringise(const MTL::HazardTrackingMode &el)
+{
+  BEGIN_ENUM_STRINGISE(MTL::HazardTrackingMode)
+  {
+    MTL_STRINGISE_ENUM(HazardTrackingModeDefault);
+    MTL_STRINGISE_ENUM(HazardTrackingModeUntracked);
+    MTL_STRINGISE_ENUM(HazardTrackingModeTracked);
+  }
+  END_ENUM_STRINGISE()
+}
+
+template <>
+rdcstr DoStringise(const MTL::ResourceOptions &el)
+{
+  BEGIN_BITFIELD_STRINGISE(MTL::ResourceOptions)
+  {
+    // MTLResourceCPUCacheModeDefaultCache
+    // MTLResourceStorageModeShared
+    // MTLResourceHazardTrackingModeDefault
+    // are all the same value (0)
+    STRINGISE_BITFIELD_CLASS_BIT_NAMED(ResourceCPUCacheModeDefaultCache,
+                                       "MTLResourceCPUCacheModeDefaultCache | "
+                                       "MTLResourceStorageModeShared | "
+                                       "MTLResourceHazardTrackingModeDefault");
```

Comment:
Unfortunately I don't think this will work. Consider if only `ResourceCPUCacheModeWriteCombined` is set - we would want a stringise result of `ResourceCPUCacheModeWriteCombined | MTLResourceStorageModeShared | MTLResourceHazardTrackingModeDefault` for consistency, but because the '0' bits won't be processed in that case so you'll just get `ResourceCPUCacheModeWriteCombined` on its own.

The way this bitfield is packed I think it makes more sense to do this stringise somewhat manually, so you can stringise each sub-section of the bitfield individually and combine it.